### PR TITLE
fix:2215 - Fixed the validation for lighthouse urls

### PR DESCRIPTION
**Pull request by riginoommen**
# Fixes 2215

### 

# Explain the fix 2215

Fixed the URL validation issues for the lighthouse spa
## Does this PR introduce a breaking change

No

## Screenshots
<details>
<summary>View Screenshots</summary>

image
image
image

</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Does the change have appropriate unit tests?
- [x] Did tests pass?

### CODE REVIEW

**Comment by riginoommen** [comment]
Fixed the above alerts

File: packages/lighthouse-spa/src/app/shared/directive/url-validator.directive.ts [hunk 1]
```diff
@@ -21,8 +22,11 @@ export class UrlValidatorDirective {
     return (control: FormControl) => {
       const url = control.value;
       try {
-        new URL(url);
-        return null;
+        if (yup.string().url().isValidSync(url) === true) {
```
Comment (deshmukhmayur) [inline packages/lighthouse-spa/src/app/shared/directive/url-validator.directive.ts:25]:
```suggestion
        if (yup.string().url().isValidSync(url)) {
```

**Review by deshmukhmayur — CHANGES_REQUESTED** [review]

File: packages/lighthouse-spa/src/app/shared/directive/url-validator.directive.ts (hunk 1 above)
Comment (riginoommen) [reply packages/lighthouse-spa/src/app/shared/directive/url-validator.directive.ts]: Updated

**Review by deshmukhmayur — APPROVED** [review]

**Review by hybridx — DISMISSED** [review]

**Comment by riginoommen** [comment]
Rebased the PR and fixed the conflicts

**Pull request merged**